src/ai_hawk/job_manager.py

```diff
@@ -354,11 +366,10 @@ def apply_jobs(self):
 
             # Continue with the job application process regardless of the applicants count check
             """
-        
 
             if self.is_previously_failed_to_apply(job.link):
                 logger.debug(f"Previously failed to apply for {job.title} at {job.company}, skipping...")
-                continue
+                # continue
```

Comment:
why was this logic change required?

If the intention is not to skip previously failed, suggest removing checks, then removing continue.
 
 I believe failed, are likely to fail again. let stick to existing one.
 
 you need to add retry code (number of retries); else it might go infinite lop (out of scope of hotfix, use release branch for this change)

Reply:
For now, I am keeping as is then